p2p/src/lib.rs

```diff
@@ -45,44 +45,34 @@ pub mod boilerplate {
 }
 
 /// Errors used in [`crate`].
-#[derive(Debug, Error)]
+#[derive(Debug, Error, displaydoc::Display)]
 pub enum Error {
-    /// Failed to read or write
-    #[error("Failed IO operation: {0}.")]
+    /// Failed IO operation
     Io(#[source] std::sync::Arc<io::Error>),
```

Comment:
I wonder why it's `Arc`, but ok

Reply:
To be cloneable. If the structure wasn't private, I'd actually look into the underlying issue, because an `IO` error could be raised in multiple situations. Unfortunately, the standard library doesn't want that use case and an `IO` error can only be parsed as text.